lldb/source/Utility/Stream.cpp

```diff
@@ -70,6 +72,31 @@ size_t Stream::PutCString(llvm::StringRef str) {
   return bytes_written;
 }
 
+void Stream::PutCStringColorHighlighted(llvm::StringRef text,
+                                        const char *pattern) {
+  if (!pattern) {
+    PutCString(text);
+    return;
+  }
+
+  // If pattern is not nullptr, we should use color
+  llvm::Regex reg_pattern(pattern);
+  llvm::SmallVector<llvm::StringRef, 1> matches;
+  llvm::StringRef remaining = text;
+  std::string format_str = lldb_private::ansi::FormatAnsiTerminalCodes(
+      "${ansi.fg.red}%.*s${ansi.normal}");
```

Comment:
Most (all?) other colors are configurable. I think this one should be too. Some examples of this are the `show-autosuggestion-ansi-prefix` and `show-progress-ansi-prefix`. This pattern should work here as well.

Reply:
I'm not sure if I got this suggestion right. What I did was:

1. Defined these properties in `lldb/source/Core/CoreProperties.td`:

```
def ShowRegexMatchAnsiPrefix: Property<"show-regex-match-ansi-prefix", "String">,
    Global,
    DefaultStringValue<"${ansi.fg.red}">,
    Desc<"When displaying a regex match in a color-enabled terminal, use the ANSI terminal code specified in this format immediately before the match.">;
def ShowRegexMatchAnsiSuffix: Property<"show-regex-match-ansi-suffix", "String">,
    Global,
    DefaultStringValue<"${ansi.normal}">,
    Desc<"When displaying a regex match in a color-enabled terminal, use the ANSI terminal code specified in this format immediately after the match.">;
```

2. Declared the getter functions in `Debugger.h` and defined them in `Debugger.cpp` following the pattern:

```cpp
llvm::StringRef Debugger::GetRegexMatchAnsiPrefix() const {
  const uint32_t idx = ePropertyShowRegexMatchAnsiPrefix;
  return GetPropertyAtIndexAs<llvm::StringRef>(
      idx, g_debugger_properties[idx].default_cstr_value);
}

llvm::StringRef Debugger::GetRegexMatchAnsiSuffix() const {
  const uint32_t idx = ePropertyShowRegexMatchAnsiSuffix;
  return GetPropertyAtIndexAs<llvm::StringRef>(
      idx, g_debugger_properties[idx].default_cstr_value);
}

```

3. Added two parameters to `Stream::PutCStringColorHighlighted`: `prefix` and `suffix`.
4. In `Address.cpp` and `Symbol.cpp`, there was a `Target` object that I used to get the debugger and its properties. In `CommandObjectTarget.cpp`, there was an `CommandInterpreter` object that was used for the same. But in `SymbolContext.cpp`, there was a `TargetSP` object and I tried using it to get the debugger properties, I got a segmentation fault, that's why I put the following two lines hard coded (while we don't solve this):

```
llvm::StringRef ansi_prefix = "${ansi.fg.red}";
llvm::StringRef ansi_suffix = "${ansi.normal}";
```

Please let me know if there's a better way to do this or if I completely misunderstood the suggestion.

Reply:
Also, I was going to add a test case for different colors, but I will only add when that error is fixed.

Reply:
Actually, before pushing these, `check-lldb` failed because of this change.

You can check the log here: [https://pastebin.com/atfvLauV](https://pastebin.com/atfvLauV)

Reply:
I think it's either find another way to get to the settings, or assume no colour if the target isn't present.

Jonas may know a way to do it.

Reply:
What are your thoughts on this approach? (In `SymbolContext.cpp` - line 98). I tested it and it only fails on the new test this PR creates, so we would only need to adapt it.

```cpp
if (name) {
        llvm::StringRef ansi_prefix;
        llvm::StringRef ansi_suffix;
        if (target_sp) {
          ansi_prefix = target_sp->GetDebugger().GetRegexMatchAnsiPrefix();
          ansi_suffix = target_sp->GetDebugger().GetRegexMatchAnsiSuffix();
        }
        s->PutCStringColorHighlighted(name.GetStringRef(), pattern, ansi_prefix,
                                      ansi_suffix);
      }
```

Or even forcing the red color when there is no target? Honestly, I do not completely understand what would mean not having a target.

Reply:
I assume if we have no target set we won't know whether to colour anything, so we'd default to no colours. Which means the prefix/suffix can default to empty string. Which is exactly what you've suggested.

Worst case in some situation it doesn't highlight and if the user cares they can open an issue for it.

I expect the SymbolContext can live without a target in some scenario, if you want to find out what, find somewhere in SymbolContext that null checks target, remove the check and see what tests fail.

Reply:
![image](https://github.com/llvm/llvm-project/assets/69206952/a9c34f6f-b671-4fc2-bc6c-abf8d9d489d2)

This is how it is working now. Basically no color for SymbolContext. IMO, it is a little weird, but the alternative option of forcing red color when no target is available could be weirder when the color was changed by the user.